virtual_ecosystem/models/abiotic/abiotic_model.py

```diff
@@ -0,0 +1,347 @@
+            .indexes["layers"]
+        )
+        true_canopy_layers_n = len(true_canopy_indexes)
+        empty_canopy_layers = self.layer_structure.canopy_layers - true_canopy_layers_n
+        topsoil_layer_index = self.layer_structure.layer_roles.index("soil")
+
+        # Wind profiles
+        wind_update_inputs: dict[str, DataArray] = {}
+
+        for var in ["layer_heights", "leaf_area_index", "air_temperature"]:
+            selection = (
+                self.data[var]
+                .where(self.data[var].layer_roles != "soil")
+                .dropna(dim="layers")
+            )
+            wind_update_inputs[var] = selection
+
+        wind_update = wind.calculate_wind_profile(
+            canopy_height=self.data["canopy_height"].to_numpy(),
+            wind_height_above=(self.data["canopy_height"] + 15).to_numpy(),
+            wind_layer_heights=wind_update_inputs["layer_heights"].to_numpy(),
+            leaf_area_index=wind_update_inputs["leaf_area_index"].to_numpy(),
+            air_temperature=wind_update_inputs["air_temperature"].to_numpy(),
+            atmospheric_pressure=self.data["atmospheric_pressure"].to_numpy()[0],
+            sensible_heat_flux_topofcanopy=(
+                self.data["sensible_heat_flux_topofcanopy"].to_numpy()
+            ),
+            wind_speed_ref=(
+                self.data["wind_speed_ref"].isel(time_index=time_index).to_numpy()
+            ),
+            wind_reference_height=(self.data["canopy_height"] + 10).to_numpy(),
+            abiotic_constants=self.model_constants,
+            core_constants=self.core_constants,
+        )  # TODO wind height above in constants, cross-check with reference heights
+
+        wind_output = {}
+        wind_speed_above_canopy = DataArray(
+            wind_update["wind_speed_above_canopy"],
+            dims="cell_id",
+            coords={"cell_id": self.data.grid.cell_id},
+        )
+        wind_output["wind_speed_above_canopy"] = wind_speed_above_canopy
+
+        for var in ["wind_speed_canopy", "friction_velocity"]:
+            var_out = DataArray(
+                np.concatenate(
+                    (
+                        wind_update[var][0 : (true_canopy_layers_n + 1)],
+                        np.full((empty_canopy_layers, self.data.grid.n_cells), np.nan),
+                        wind_update[var][
+                            (true_canopy_layers_n + 1) : (len(wind_update[var]) + 1)
+                        ],
+                        np.full(
+                            (
+                                len(self.layer_structure.soil_layers),
+                                self.data.grid.n_cells,
+                            ),
+                            np.nan,
+                        ),
+                    ),
+                ),
+                dims=self.data["layer_heights"].dims,
+                coords=self.data["layer_heights"].coords,
+            )
+            wind_output[var] = var_out
```

Comment:
So the same basic pattern would be good here too:

```suggestion
        # I'm sure there are better ways of getting this index and it may be used elsewhere too
        true_data_rows = list(range(0, true_canopy_layers_n + 1)) + list(
            range(
                true_canopy_layers_n + empty_canopy_layers + 1,
                true_canopy_layers_n + empty_canopy_layers + 3,
            )
        )

        for var in ["wind_speed_canopy", "friction_velocity"]:
            # Might make sense to store the shape somewhere and just use np.full(shape, np.nan)
            var_data = np.full_like(self.data['leaf_area_index'], np.nan)
            var_data[true_data_rows, :] = wind_update[var]
            var_out = DataArray(
                var_data,
                dims=self.data["layer_heights"].dims,
                coords=self.data["layer_heights"].coords,
            )
            wind_output[var] = var_out
```